src/node/media_stream_destination.rs

```diff
@@ -0,0 +1,163 @@
+/// - see also: [`BaseAudioContext::create_media_stream_destination`](crate::context::BaseAudioContext::create_media_stream_destination)
+///
+/// Since the w3c `MediaStream` interface is not part of this library, we cannot adhere to the
+/// official specification. Instead, you can pass in any callback that handles audio buffers.
+///
+/// IMPORTANT: you must consume the buffers faster than the render thread produces them, or you
+/// will miss frames. Consider to spin up a dedicated thread to consume the buffers and cache them.
+///
+/// # Usage
+///
+/// ```no_run
+/// use web_audio_api::context::{AudioContext, BaseAudioContext};
+/// use web_audio_api::node::{AudioNode, AudioScheduledSourceNode};
+///
+/// // Create an audio context where all audio nodes lives
+/// let context = AudioContext::new(None);
+///
+/// // Create an oscillator node with sine (default) type
+/// let osc = context.create_oscillator();
+///
+/// // Create a media destination node
+/// let dest = context.create_media_stream_destination();
+/// osc.connect(&dest);
+/// osc.start();
+///
+/// // Handle recorded buffers
+/// println!("samples recorded:");
+/// let mut samples_recorded = 0;
+/// for item in dest.stream() {
+///     let buffer = item.unwrap();
+///
+///     // You could write the samples to a file here.
+///     samples_recorded += buffer.length();
+///     print!("{}\r", samples_recorded);
+/// }
+/// ```
+///
+/// # Examples
+///
+/// - `cargo run --release --example recorder`
+
+pub struct MediaStreamAudioDestinationNode {
+    registration: AudioContextRegistration,
+    channel_config: ChannelConfig,
+    receiver: Receiver<AudioBuffer>,
+}
+
+impl AudioNode for MediaStreamAudioDestinationNode {
+    fn registration(&self) -> &AudioContextRegistration {
+        &self.registration
+    }
+
+    fn channel_config_raw(&self) -> &ChannelConfig {
+        &self.channel_config
+    }
+
+    fn number_of_inputs(&self) -> u32 {
+        1
+    }
+
+    fn number_of_outputs(&self) -> u32 {
+        0
+    }
+}
+
+impl MediaStreamAudioDestinationNode {
+    /// Create a new MediaStreamAudioDestinationNode
+    pub fn new<C: BaseAudioContext>(context: &C, options: ChannelConfigOptions) -> Self {
+        context.base().register(move |registration| {
+            let (send, recv) = crossbeam_channel::bounded(1);
+            let recv_control = recv.clone();
+
+            let node = MediaStreamAudioDestinationNode {
+                registration,
+                channel_config: options.into(),
+                receiver: recv_control,
+            };
+
+            let render = DestinationRenderer { send, recv };
+
+            (node, Box::new(render))
+        })
+    }
+
+    /// A [`MediaStream`] iterator producing audio buffers with the same number of channels as the
+    /// node itself
+    ///
+    /// While you could call this function multiple times and poll all iterators concurrently, each
+    /// buffer will only be offered once.
```

Comment:
??

Reply:
Check the example after 
> Note that cloning only creates a new handle to the same sending or receiving side. It does not create a separate stream of messages in any way:
At https://docs.rs/crossbeam-channel/latest/crossbeam_channel/index.html
Items can only be received once

Reply:
ok that's what I imagined, maybe your comment should be something like?

```
    /// Note that while you can call this function multiple times and poll all iterators concurrently, 
    /// this could lead to unexpected behavior as the buffers will only be offered once.
```

a bit more like a warning: "don't do that!"

Reply:
done